packages/react/src/primitives/shared/styleUtils.ts

```diff
@@ -44,31 +49,43 @@ export const convertGridSpan = (
   }
   // ResponsiveObject<PropertyType>
   if (typeof spanValue === 'object' && spanValue != null) {
-    const newObj: ResponsiveObject<string> = {};
-    Object.entries(spanValue).forEach(([key, value]) => {
-      newObj[key] = getGridSpan(value as GridSpanType);
-    });
+    let newObj: ResponsiveObject<string> = {};
+    Object.entries(spanValue).forEach(
+      ([key, value]: [string, GridSpanType]) => {
+        newObj = {
+          ...newObj,
+          [key]: getGridSpan(value),
+        };
```

Comment:
For some reason TS likes doing this spread object thing but not `newObj[key] = getGridSpan(value)` 🤷